python/langsmith/client.py

```diff
@@ -1293,6 +1278,7 @@ def create_run(
             if self._pyo3_client is not None:
                 self._pyo3_client.create_run(run_create)
             elif self.compressed_runs is not None:
+                assert self._data_available_event is not None
```

Comment:
Could we have useful error messages if these trip?

Reply:
agree ^, we should avoid asserts in user-facing code